web-search/web-search/src/durability.rs

```diff
@@ -0,0 +1,359 @@
+                        ))
+                    }
+                };
+
+                result
+            } else {
+                println!("[DURABILITY] next_page: REPLAY mode - retrieving persisted result");
+                let result: Result<SearchResult, SearchError> = durability.replay_infallible();
+                
+                // Update state during replay to maintain consistency
+                let mut state = self.state.borrow_mut();
+                match &mut *state {
+                    Some(DurableSearchSessionState::Replay { 
+                        page_count, 
+                        finished, 
+                        .. 
+                    }) => {
+                        match &result {
+                            Ok(_) => {
+                                // Increment page count on successful replay
+                                *page_count += 1;
+                                println!("[DURABILITY] next_page: REPLAY mode - incremented page_count to: {}", *page_count);
+                            }
+                            Err(_) => {
+                                // Mark as finished on error
+                                *finished = true;
+                                println!("[DURABILITY] next_page: REPLAY mode - marked as finished due to error");
+                            }
+                        }
+                    }
+                    _ => {
+                        // This shouldn't happen during replay mode
+                        println!("[DURABILITY] next_page: REPLAY mode - unexpected state");
+                    }
+                }
+                
+                match result {
+                    Ok(search_result) => {
+                        println!(
+                            "[DURABILITY] next_page: REPLAY mode - replayed result: {}",
+                            search_result.title
+                        );
+                        Ok(search_result)
+                    }
+                    Err(error) => {
+                        println!(
+                            "[DURABILITY] next_page: REPLAY mode - replayed error: {:?}",
+                            error
+                        );
+                        Err(error)
+                    }
+                }
+            }
+        }
+
+        fn get_metadata(&self) -> Option<SearchMetadata> {
+            let durability = Durability::<Option<SearchMetadata>, UnusedError>::new(
+                "golem_web_search",
+                "get_metadata", 
+                DurableFunctionType::ReadRemote,
+            );
+
+            if durability.is_live() {
+                println!("[DURABILITY] get_metadata: LIVE mode - retrieving metadata");
+                let state = self.state.borrow();
+                let result = match &*state {
+                    Some(DurableSearchSessionState::Live { session, .. }) => session.get_metadata(),
+                    _ => None,
```

Comment:
This case needs to be handled (we just finished replay and the user asks us for the current metadata).